perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AuthzResolverBlImpl.java

```diff
@@ -1210,6 +1187,11 @@ public static synchronized void refreshAuthz(PerunSession sess) throws InternalE
 			}
 		}
 
+		//for OAuth clients, do not allow delegating PERUNADMIN role
```

Comment:
Regarding fixing above impl, this should be right after calling ``prepareServiceRoles(sess);`` and before check if roles are non-empty and considered initialized. Otherwise you would probably get perun_admin even for OAUTH client.